packages/components/src/VirtualizedList/CellTitle/CellTitle.component.js

```diff
@@ -47,6 +49,19 @@ class CellTitle extends React.Component {
 		const titleId = id && `${id}-${rowIndex}-title-cell`;
 		const actionsId = id && `${id}-${rowIndex}-title-actions`;
 
+		let icon = null;
+		if (iconKey) {
+			if (iconTooltipLabelKey && rowData[iconKey] && rowData[iconTooltipLabelKey]) {
+				icon = (
+					<TooltipTrigger label={rowData[iconTooltipLabelKey]} tooltipPlacement="top">
+						<Icon name={rowData[iconKey]} className={theme.icon} />
+					</TooltipTrigger>
+				);
+			} else if (rowData[iconKey]) {
+				icon = <Icon name={rowData[iconKey]} className={theme.icon} />;
```

Comment:
To avoid this duplication (and it removes 1 level of complexity)
```
let icon;
if (iconKey && rowData[iconKey]) {
	icon = ...
}
if (icon && iconTooltipLabelKey && rowData[iconTooltipLabelKey]) {
	icon = <TooltipTrigger ...>{icon}</>
}
```